keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/ProposalApplicationStrategy.java

```diff
@@ -0,0 +1,21 @@
+package io.improbable.keanu.algorithms.mcmc;
+
+import io.improbable.keanu.algorithms.mcmc.proposal.Proposal;
+import io.improbable.keanu.algorithms.variational.optimizer.Variable;
+
+import java.util.Set;
+
+public interface ProposalApplicationStrategy {
+
+    /**
+     * Any actions taken after after applying the provided proposal
+     * from a sampling algorithm to its variables.
+     *
+     * No action will be taken in most cases.
+     *
+     * @param proposal  the proposal that has been applied
```

Comment:
`proposal` is never used... have you put it in with a view to what information someone might need if they wanted to write a new one?